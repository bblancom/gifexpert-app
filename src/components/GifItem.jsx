import PropTypes from 'prop-types';

export const GifItem = ({ title, url, id }) => {
   return (
      <div
         key={id}
         className="card"
      >
         <img
            src={url}
            alt={title}
         />
         <p>{title}</p>
      </div>
   );
};

GifItem.propTypes = {
   title: PropTypes.string.isRequired,
   url: PropTypes.string.isRequired,
   id: PropTypes.string,
};
