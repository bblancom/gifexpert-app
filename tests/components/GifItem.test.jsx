import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Tests on GifItem component', () => {
   const title = 'Saitama';
   const url = 'https://one-punch.com/saitama.jpg';

   test('Should match the snapshot', () => {
      const { container } = render(
         <GifItem
            title={title}
            url={url}
         />
      );
      expect(container).toMatchSnapshot();
   });

   test('Picture is displayed with the right url and alt', () => {
      render(
         <GifItem
            title={title}
            url={url}
         />
      );
      const { src, alt } = screen.getByRole('img');
      expect(src).toBe(url);
      expect(alt).toBe(title);
   });

   test('Displays title on component', () => {
      render(
         <GifItem
            title={title}
            url={url}
         />
      );
      expect(screen.getByText(title)).toBeTruthy();
   });
});
