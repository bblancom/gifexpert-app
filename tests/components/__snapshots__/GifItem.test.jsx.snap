// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifItem component Should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
